yt_dlp/downloader/mhtml.py

```diff
@@ -0,0 +1,175 @@
+# coding: utf-8
+from __future__ import unicode_literals
+from .fragment import FragmentFD
+import uuid
+import io
+from ..version import __version__ as YT_DLP_VERSION
+from ..utils import (
+    urljoin,
+    srt_subtitles_timecode,
+    formatSeconds,
+)
+
+
+class MhtmlFD(FragmentFD):
+    FD_NAME = 'mhtml'
+    _EXTENSION = 'mhtml'
+
+    _STYLESHEET = """\
+html, body {
+    margin: 0;
+    padding: 0;
+    height: 100vh;
+}
+
+html {
+    overflow-y: scroll;
+    scroll-snap-type: y mandatory;
+}
+
+body {
+    scroll-snap-type: y mandatory;
+    display: flex;
+    flex-flow: column;
+}
+
+body > figure {
+    max-width: 100vw;
+    max-height: 100vh;
+    scroll-snap-align: center;
+}
+
+body > figure > figcaption {
+    text-align: center;
+    height: 2.5em;
+}
+
+body > figure > img {
+    display: block;
+    margin: auto;
+    max-width: 100%;
+    max-height: calc(100vh - 5em);
+}
+"""
+
+    def _gen_cid(self, i, fragment, frag_boundary):
+        return '%u.%s@yt-dlp.github.io.invalid' % (i, frag_boundary.decode('us-ascii'))
+
+    def _gen_stub(self, fragments, frag_boundary):
+        output = io.StringIO()
+
+        output.write(
+            '<!DOCTYPE html>'
+            '<html>'
+            '<head>'
+            ''  '<meta name="generator" content="yt-dlp %s">'
+            ''  '<title>yt-dlp downloaded slideshow</title>'
+            ''  '<style>%s</style>'
+            '<body>'
+            '\n'
+            % (
+                YT_DLP_VERSION,
+                self._STYLESHEET,
+            )
+        )
+
+        t0 = 0
+        for i, frag in enumerate(fragments):
+            t1 = t0 + frag['duration']
+            output.write(
+                '<figure>'
+                    '<figcaption>Slide #%u: %s – %s (duration: %s)</figcaption>'
+                    '<img src="cid:%s">'
+                '</figure>\n'
+                % (
+                    i + 1,
+                    srt_subtitles_timecode(t0),
+                    srt_subtitles_timecode(t1),
+                    formatSeconds(frag['duration']),
```

Comment:
Why?

Reply:
When I was testing, I came across slides with `0` duration. This is why I thought of this. This honestly does not matter much, but why not add the msec since we have the data available and you are putting it in the timecode anyway

Reply:
Oh, duration *rounded* to zero. Well, I figure it might make sense, but like I said before, I didn’t want to spend too much time on it and there wasn’t a ready method for it.